py/desitarget/cuts.py

```diff
@@ -1910,6 +1910,8 @@ def isQSO_randomforest(gflux=None, rflux=None, zflux=None, maskbits=None,
     if maskbits is not None:
         # ADM default mask bits from the Legacy Surveys not set.
         preSelection &= imaging_mask(maskbits)
+        
+    preSelection = preSelection.astype('bool')
```

Comment:
Following discussion in the PR, I suggest that we drop this `preSelection` type casting line and instead around line 1876 update it to be
```python
if primary is None:
    primary = np.ones_like(gflux, dtype=bool)
else:
    assert primary.dtype == np.dtype(bool)
```
or something similar to check that `primary` is an array of booleans.

And also update the docstring to be complete with describing the inputs and their types.  e.g. it is also unclear to me if this function is supposed to support scalar inputs for testing individual targets, or whether it only works with input arrays and scalars are explicitly not supported.

@geordie666 ?